src/commands/runFromZipDeploy.ts

```diff
@@ -0,0 +1,211 @@
+            name = storageAccount.name;
+            key = result.keys[0].value;
+        }
+    }
+
+    if (name !== undefined && key !== undefined) {
+        return azureStorage.createBlobService(name, key);
+    } else {
+        throw new ArgumentError({ name: name, key: key });
+    }
+}
+
+async function createBlobFromZip(blobService: azureStorage.BlobService, zipFilePath: string, blobName: string): Promise<string> {
+    const containerName: string = 'azureappservice-run-from-zip';
+    await new Promise<void>((resolve: () => void, reject: (err: Error) => void): void => {
+        blobService.createContainerIfNotExists(containerName, (err: Error) => {
+            if (err !== undefined) {
+                reject(err);
+            } else {
+                resolve();
+            }
+        });
+    });
+
+    await new Promise<void>((resolve: () => void, reject: (err: Error) => void): void => {
+        blobService.createBlockBlobFromLocalFile(containerName, blobName, zipFilePath, (error: Error, _result: azureStorage.BlobService.BlobResult, _response: azureStorage.ServiceResponse) => {
+            if (error !== undefined) {
+                // tslint:disable-next-line:no-any
+                const errorAny: any = error;
+                // tslint:disable-next-line:no-unsafe-any
+                if (!!errorAny.code) {
+                    // tslint:disable-next-line:no-unsafe-any
+                    let humanReadableMessage: string = `Unable to save '${blobName}', blob service returned error code "${errorAny.code}"`;
+                    // tslint:disable-next-line:no-unsafe-any
+                    switch (errorAny.code) {
+                        case 'ENOTFOUND':
+                            humanReadableMessage += ' - Please check connection.';
+                            break;
+                        default:
+                            break;
+                    }
+                    reject(new Error(humanReadableMessage));
+                } else {
+                    reject(error);
+                }
+            } else {
+                resolve();
+            }
+        });
+    });
+    const sasToken: string = blobService.generateSharedAccessSignature(containerName, blobName, <azureStorage.common.SharedAccessPolicy>{
+        AccessPolicy: {
+            Permissions: azureStorage.BlobUtilities.SharedAccessPermissions.READ + azureStorage.BlobUtilities.SharedAccessPermissions.LIST,
+            Start: azureStorage.date.secondsFromNow(-10),
+            // for clock desync
+            Expiry: azureStorage.date.daysFromNow(365),
+            ResourceTypes: azureStorage.BlobUtilities.BlobContainerPublicAccessType.BLOB
+        }
+    });
+
+    return blobService.getUrl(containerName, blobName, sasToken, true);
+}
+
+export async function isDirectory(fsPath: string): Promise<boolean> {
+    const fsStats: fs.Stats = await new Promise((resolve: (s?: fs.Stats) => void, reject: (e: Error) => void): void => {
```

Comment:
Not sure this is worth it's own function since it can be done in one line:
```
(await fse.lstat(fsPath)).isDirectory()
```

But if you do keep it as its own function, don't export it from this file